caffe2/operators/distance_op.cu

```diff
@@ -316,7 +316,7 @@ bool CosineSimilarityOp<float, CUDAContext>::RunOnDevice() {
   const float* X_data = X.data<float>();
   const float* Y_data = Y.data<float>();
   // Auxiliary arrays, one allocation of memory
-  aux_.Resize(2 * N);
+  ReinitializeTensor(&aux_, {2 * N}, at::dtype<float>().device(CUDA));
```

Comment:
Isn't the specification of CUDA here supposed to be unnecessary, because we should infer the device type from the Operator context?

Reply:
ReinitializeTensor is a free function, we can use it outside of Operator class as well. Also we can have Tensor{CPU} even when context is CUDA, so it's not necessarily equivalent to the Context.
This is a natural codemod from the declaration of "Tensor x{CUDA};" We can have a special wrapper for it inside Operator though.